scripts/cmake/version.cmake

```diff
@@ -59,6 +59,27 @@ if(NOT SOF_TAG)
 	set(SOF_TAG 0)
 endif()
 
+# list tracked files from src directory
+execute_process(COMMAND git ls-files src
+		WORKING_DIRECTORY ${SOF_ROOT_SOURCE_DIRECTORY}
+		OUTPUT_FILE tracked_file_list
+	)
+# get hash of each listed files
+execute_process(COMMAND git hash-object --stdin-paths
+		WORKING_DIRECTORY ${SOF_ROOT_SOURCE_DIRECTORY}
+		INPUT_FILE tracked_file_list
+		OUTPUT_FILE tracked_file_hash_list
+	)
+# then calculate single hash of previously calculated hash list
+execute_process(COMMAND git hash-object --stdin
+		WORKING_DIRECTORY ${SOF_ROOT_SOURCE_DIRECTORY}
+		OUTPUT_STRIP_TRAILING_WHITESPACE
+		INPUT_FILE tracked_file_hash_list
+		OUTPUT_VARIABLE SOF_FW_HASH_LONG
+	)
+string(SUBSTRING ${SOF_FW_HASH_LONG} 0 8 SOF_FW_HASH)
+message(STATUS "Firmware content hash: ${SOF_FW_HASH}")
+
```

Comment:
Or on a different line: how can one identify a firmware based on a hash value? if I get 0xdeadbeef from a customer log, what's the next step? Can I find out what this firmware was from? Or is this enough to have the last git commit and a + that identifies the presence of local changes?

Reply:
I am not sure what is the purpose of logging the hash value in dma_trace_buffer_init(). The hash helps to verify integrity of trace content and the ldc file used to decode the traces). SOF_TAG is sufficient for identification if the fw is built directly from a commit w/o local changes. SOF_FW_HASH value reflects local changes so using it for log decoding is more reliable in the development process. I'd prefer to have SOF_TAG logged as the first trace line (while still use HASH for .ldc integrity check).

Reply:
> I am not sure what is the purpose of logging the hash value in dma_trace_buffer_init(). The hash helps to verify integrity of trace content and the ldc file used to decode the traces). SOF_TAG is sufficient for identification if the fw is built directly from a commit w/o local changes. SOF_FW_HASH value reflects local changes so using it for log decoding is more reliable in the development process. I'd prefer to have SOF_TAG logged as the first trace line (while still use HASH for .ldc integrity check).

All I care about is to have a well-defined/repeatable pattern that can be used to figure out if the trace works or not.

Reply:
>Or is this enough to have the last git commit and a + that identifies the presence of local changes?

This one, but git commit hash must be red from extended manifest or fw_ready, so maybe I should add this value to log message, to have everything in one place

Reply:
>I am not sure what is the purpose of logging the hash value in dma_trace_buffer_init(). The hash helps to verify integrity of trace content and the ldc file used to decode the traces). SOF_TAG is sufficient for identification if the fw is built directly from a commit w/o local changes. SOF_FW_HASH value reflects local changes so using it for log decoding is more reliable in the development process. I'd prefer to have SOF_TAG logged as the first trace line (while still use HASH for .ldc integrity check).

After fw run, we have logger executable, ldc file, fw_ready file and binary trace data data to get user readable form of traces. Compatibility between logger and ldc file is checked by DBG_ABI, between fw_ready and ldc file by source hash, but there wasn't any check (or even saved information) between ldc/fw_ready and binary trace files. Logging FW source code hash value add possibility to check this compatibility without major DBG_ABI bump.

Reply:
@ktrzcinx please handle traball case, thanks

Reply:
>I'm not sure if we should have so many tmp files in root of build folder

Ok, so I remove temp files after final hash number extraction